core/log/logger.go

```diff
@@ -23,11 +31,18 @@ type Logger interface {
 	// The key of the tuple must be a string.
 	Debug(msg string, keyVals ...any)
 
-	// With returns a new wrapped logger with additional context provided by a set.
-	With(keyVals ...any) Logger
-
 	// Impl returns the underlying logger implementation.
 	// It is used to access the full functionalities of the underlying logger.
 	// Advanced users can type cast the returned value to the actual logger.
 	Impl() any
 }
+
+// LoggerV2 is the newer Cosmos SDK logger interface. It can be used in libraries without any need to import
+// the core or log packages directly.
+type LoggerV2 interface {
+	LoggerBase
+
+	// WithV2 returns a type which can be cast to LoggerV2 with additional context provided by a set.
+	// any is returned instead of LoggerV2 to avoid the need for log users to import the log package directly.
+	WithV2(keyVals ...any) any
+}
```

Comment:
I'm not attached to this naming if someone can think of something better.

Reply:
AH! I was scratching my head yesterday about how to achieve the same thing https://github.com/cosmos/cosmos-sdk/compare/main...julien/log-dep :D

We quite need `log` to have no `core` dep indeed.

Reply:
I actually don't like `WithV2` in a function name if this is going to be long lived. How about `WithContext`?